src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -63,5 +67,17 @@ export const walletSupportsChain: UseWalletSupportsChain = ({ chainId, wallet })
   }
 }
 
+export const useWalletSupportsBuyAsset = () => {
+  const buyAssetChainId = useSwapperStore(selectBuyAsset).chainId
+  const wallet = useWallet().state.wallet
+  return walletSupportsChain({ chainId: buyAssetChainId, wallet })
+}
+
+export const useWalletSupportsSellAsset = () => {
+  const sellAssetChainId = useSwapperStore(selectSellAsset).chainId
+  const wallet = useWallet().state.wallet
+  return walletSupportsChain({ chainId: sellAssetChainId, wallet })
```

Comment:
i wonder if there would be benefit in wrapping this in useMemo?

Reply:
I don't believe so? The two dependencies of this method are nested properties of hooks return values (which themselves are obviously reactive, because hooks), and the output is a primitive, so this would both run as many times, and invalidates as often with or without a `useMemo` for this expression.

Reply:
Removed this guy anyway in https://github.com/shapeshift/web/pull/4479/commits/5c61fcb2a885f96d99a3fb2122523a55c79412fb :)